stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/MimeTypeIconProvider.kt

```diff
@@ -1,9 +1,26 @@
-package io.getstream.chat.android.ui.common
+package io.getstream.chat.android.ui
 
 import com.getstream.sdk.chat.model.ModelType
-import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.message.input.attachment.file.internal.FileAttachmentAdapter
+import io.getstream.chat.android.ui.message.input.attachment.selected.internal.SelectedFileAttachmentAdapter
 
-public object UiUtils {
+/**
+ * A class that provides icons for file attachments.
+ *
+ * @see [FileAttachmentAdapter.FileAttachmentViewHolder]
+ * @see [SelectedFileAttachmentAdapter.SelectedFileAttachmentViewHolder]
+ */
+public fun interface MimeTypeIconProvider {
+    /**
+     * Return icon drawable resource for the given MIME type.
+     *
+     * @param mimeType mimeType A MIME type (i.e. application/pdf)
+     * @return The drawable resource for the given MIME type
+     */
+    public fun getIconRes(mimeType: String?): Int
+}
+
+public class MimeTypeIconProviderImpl : MimeTypeIconProvider {
```

Comment:
Is this impl class meant to be public?

Reply:
It should be to help our customers to only override some icons, but keeps the rest of them with our implementation